test/e2e/specs/onboarding/signup__with-theme-premium.ts

```diff
@@ -0,0 +1,218 @@
+ *
+ * Keywords: Onboarding, Store Checkout, Coupon, Signup, Plan, Subscription, Cancel
+ */
+describe( 'With theme - Premium theme: Signup, onboard, launch and cancel subscription', function () {
+	const planName = 'Premium';
+	const themeSlug = 'outland';
+
+	const testUser = DataHelper.getNewTestUser( {
+		usernamePrefix: 'ftmepersonal',
+	} );
+
+	let page: Page;
+	let newUserDetails: NewUserResponse;
+	let newSiteDetails: NewSiteResponse;
+
+	beforeAll( async () => {
+		page = await browser.newPage();
+	} );
+
+	describe( 'Signup', function () {
+		let cartCheckoutPage: CartCheckoutPage;
+		let signupPickPlanPage: SignupPickPlanPage;
+		let originalAmount: number;
+
+		beforeAll( async function () {
+			await BrowserManager.setStoreCookie( page, { currency: 'GBP' } );
+		} );
+
+		it( 'Navigate to Signup page', async function () {
+			const themeDetailsPage = new ThemesDetailPage( page );
+			await themeDetailsPage.visitTheme( themeSlug );
+
+			await themeDetailsPage.pickThisDesign();
+		} );
+
+		it( 'Sign up as new user', async function () {
+			const userSignupPage = new UserSignupPage( page );
+			newUserDetails = await userSignupPage.signupSocialFirstWithEmail( testUser.email );
+		} );
+
+		it( 'Skip domain selection', async function () {
+			const domainSearch = new DomainSearchComponent( page );
+
+			await domainSearch.search( testUser.siteName );
+			await domainSearch.selectDomain( `${ testUser.siteName }.wordpress.com` );
+		} );
+
+		it( `Select WordPress.com ${ planName } plan`, async function () {
+			signupPickPlanPage = new SignupPickPlanPage( page );
+			newSiteDetails = await signupPickPlanPage.selectPlan( planName );
+		} );
+
+		it( 'See secure payment', async function () {
+			cartCheckoutPage = new CartCheckoutPage( page );
+			await cartCheckoutPage.validateCartItem( `WordPress.com ${ planName }` );
+		} );
+
+		it( 'Prices are shown in GBP', async function () {
+			const cartAmount = ( await cartCheckoutPage.getCheckoutTotalAmount( {
+				rawString: true,
+			} ) ) as string;
+			expect( cartAmount.startsWith( '£' ) ).toBe( true );
+		} );
+
+		it( 'Apply coupon', async function () {
+			originalAmount = ( await cartCheckoutPage.getCheckoutTotalAmount() ) as number;
+			await cartCheckoutPage.enterCouponCode( SecretsManager.secrets.testCouponCode );
+		} );
+
+		it( 'Apply coupon and validate purchase amount', async function () {
+			const newAmount = ( await cartCheckoutPage.getCheckoutTotalAmount() ) as number;
+
+			expect( newAmount ).toBeLessThan( originalAmount );
+			const expectedAmount = originalAmount * 0.99;
+
+			// Some currencies do not typically have decimal places.
+			// eg. USD would commonly have 2 decimal places, e.g. 12.34.
+			// In JPY or TWD there will be no decimal digits.
+			// Drop decimals so that the result won't be affected by the currency variation.
+			expect( newAmount ).toStrictEqual( expectedAmount );
+		} );
```

Comment:
**suggestion:non-blocking** - conversely, we already test this in the other flow that you copied this from, so I don't think this is really necessary. Up to you if you'd like to remove or keep it, doesn't hurt.